### PR TITLE
WT-11339 Chunk cache pinning functional testing

### PR DESCRIPTION
**Pull request by Siddhartha8899**
I have added functionality testing for pinned chunks. Currently, it's failing as there is a issue with pinned contents, will fix the same in this PR in future commits.

### CODE REVIEW

**Comment by korteland** [comment]
In general - I like the shape of this change, once we've figured out the naming problem I don't see an issue with merging this.

File: src/block_cache/block_chunkcache.c [hunk 1]
```diff
@@ -299,14 +299,14 @@ __chunkcache_free_chunk(WT_SESSION_IMPL *session, WT_CHUNKCACHE_CHUNK *chunk)
  *     ID.
  */
 static inline uint64_t
-__chunkcache_tmp_hash(WT_CHUNKCACHE *chunkcache, WT_CHUNKCACHE_HASHID *hash_id, WT_BLOCK *block,
```
Comment (korteland) [inline src/block_cache/block_chunkcache.c:302]: Not taking the block here is a nice improvement - thanks.

File: test/suite/test_chunkcache03.py [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        self.read_and_verify(uris[1], ds[1])
+        chunks_inuse_including_pinned = self.get_stat(wiredtiger.stat.conn.chunk_cache_chunks_inuse)
+        self.assertGreater(chunks_inuse_including_pinned, chunks_inuse_excluding_pinned)
+
+        # Assert that chunks are pinned.
+        pinned_chunks_inuse = self.get_stat(wiredtiger.stat.conn.chunk_cache_chunks_pinned)
+        self.assertGreater(pinned_chunks_inuse, 0)
+
+        # Assert that the difference b/w the total chunks present and the unpinned chunks equal pinned chunks.
+        # This proves that the chunks read form pinned objects were all pinned.
```
Comment (korteland) [inline test/suite/test_chunkcache03.py:128]: Nitpick: form -> from.

**Review by korteland — APPROVED** [review]
I really like how this change turned out - nice work! LGTM.

File: test/suite/test_chunkcache03.py [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+import threading
+import time
+import wiredtiger, wttest
+
+from random import randrange
+from wtdataset import SimpleDataSet
+from wtscenario import make_scenarios
+
+'''
+- Evaluating chunkcache performance both in-memory and on-disk, to test the functionality of pinned chunks.
```
Comment (ruby-oujo) [inline test/suite/test_chunkcache03.py:40]: NP: In our test headers / comments we don't typically use continuous tense e.g evaluating -> evaluate.  Just a NP though, feel free to merge without change.

**Review by ruby-oujo — APPROVED** [review]
LGTM!

File: test/suite/test_chunkcache03.py (hunk 2 above)
Comment (Siddhartha8899) [reply test/suite/test_chunkcache03.py]: Updated in [c2195b0](https://github.com/wiredtiger/wiredtiger/pull/9555/commits/c2195b03581ef2cd05b1229fde134027b54fc589)

File: test/suite/test_chunkcache03.py (hunk 3 above)
Comment (Siddhartha8899) [reply test/suite/test_chunkcache03.py]: Updated in [c2195b0](https://github.com/wiredtiger/wiredtiger/pull/9555/commits/c2195b03581ef2cd05b1229fde134027b54fc589)

**Comment by Siddhartha8899** [comment]
evergreen merge

**Pull request merged**